### PR TITLE
Add documentation for React Router 6.4 support.

### PR DESCRIPTION
**Pull request by onurtemizkan**
Documents: https://github.com/getsentry/sentry-javascript/pull/6172

TODO: Update the version

### CODE REVIEW

File: src/platforms/javascript/guides/react/configuration/integrations/react-router.mdx [hunk 1]
```diff
@@ -34,6 +34,51 @@ Make sure `Sentry.reactRouterV6Instrumentation` is initialized by your `Sentry.i
 
 </Alert>
 
+### Usage with React Router 6.4 Data API
+_(Available in version 7.21.0 and above)_
```
Comment (AbhiPrasad) [inline src/platforms/javascript/guides/react/configuration/integrations/react-router.mdx:38]: Let's move this above v6?

File: src/platforms/javascript/guides/react/configuration/integrations/react-router.mdx [hunk 2]
```diff
@@ -34,6 +34,51 @@ Make sure `Sentry.reactRouterV6Instrumentation` is initialized by your `Sentry.i
...
+
+const router = sentryCreateBrowserRouter([
+  // ...
+]);
+
+```
+
+<Alert level="warning" title="Note">
+
+[`createHashRouter`](https://reactrouter.com/en/main/routers/create-hash-router) is not yet supported by Sentry. On the other hand [`createMemoryRouter`](https://reactrouter.com/en/main/routers/create-memory-router) can be instrumented using `wrapCreateBrowserRouter` function.
```
Comment (AbhiPrasad) [inline src/platforms/javascript/guides/react/configuration/integrations/react-router.mdx:77]: we should create a GH issue about `createHashRouter` that we can link folks to here.

File: src/platforms/javascript/guides/react/configuration/integrations/react-router.mdx (hunk 1 above)
Comment (onurtemizkan) [reply src/platforms/javascript/guides/react/configuration/integrations/react-router.mdx:38]: The initialization part above is common for all 6.x versions (`reactRouterV6Instrumentation(...)`). Will have to repeat that info if we do that?

Comment (AbhiPrasad) [reply src/platforms/javascript/guides/react/configuration/integrations/react-router.mdx:38]: Ah fair, than we can keep it here.

File: src/platforms/javascript/guides/react/configuration/integrations/react-router.mdx [hunk 3]
```diff
@@ -34,6 +34,51 @@ Make sure `Sentry.reactRouterV6Instrumentation` is initialized by your `Sentry.i
 
 </Alert>
 
+### Usage with React Router 6.4 Data API
+_(Available in version 7.21.0 and above)_
+
+
+If you create your router instance with [`createBrowserRouter`](https://reactrouter.com/en/main/routers/create-browser-router) from `react-router-dom` package, you can use `Sentry.wrapCreateBrowserRouter` to wrap it with the instrumentation:
+
```
Comment (lizokm) [inline src/platforms/javascript/guides/react/configuration/integrations/react-router.mdx:42]:
```suggestion
If you choose to create your router instance with [`createBrowserRouter`](https://reactrouter.com/en/main/routers/create-browser-router) from the `react-router-dom` package, you can use `Sentry.wrapCreateBrowserRouter` to wrap it with the instrumentation:

```

File: src/platforms/javascript/guides/react/configuration/integrations/react-router.mdx [hunk 4]
```diff
@@ -34,6 +34,51 @@ Make sure `Sentry.reactRouterV6Instrumentation` is initialized by your `Sentry.i
...
+const router = sentryCreateBrowserRouter([
+  // ...
+]);
+
+```
+
+<Alert level="warning" title="Note">
+
+[`createHashRouter`](https://reactrouter.com/en/main/routers/create-hash-router) is not yet supported by Sentry. On the other hand [`createMemoryRouter`](https://reactrouter.com/en/main/routers/create-memory-router) can be instrumented using `wrapCreateBrowserRouter` function.
```
Comment (lizokm) [inline src/platforms/javascript/guides/react/configuration/integrations/react-router.mdx:77]:
```suggestion
While [`createHashRouter`](https://reactrouter.com/en/main/routers/create-hash-router) isn't supported by Sentry yet, you can instrument [`createMemoryRouter`](https://reactrouter.com/en/main/routers/create-memory-router) using the `wrapCreateBrowserRouter` function.
```

**Review by lizokm — APPROVED** [review]
Made a few small tweaks, but otherwise this looks good.

**Review by AbhiPrasad — APPROVED** [review]
Released in `7.21.0` so merging https://github.com/getsentry/sentry-javascript/blob/master/CHANGELOG.md#7210

**Pull request merged**